File: src/main/java/com/dozycoffee/wms/warehouse/domain/enumeration/ZoneCode.java
package com.dozycoffee.wms.warehouse.domain.enumeration;

import com.dozycoffee.wms.warehouse.domain.valueobject.Capacity;
import lombok.Getter;

@Getter
public enum ZoneCode {

    A("원두", TemperatureType.AMBIENT, new Capacity(180)),
    B("시럽", TemperatureType.AMBIENT, new Capacity(120)),
    C("분말/파우더", TemperatureType.AMBIENT, new Capacity(100)),
    D("유제품", TemperatureType.COLD, new Capacity(80)),
    E("컵/소모품/포장재", TemperatureType.AMBIENT, new Capacity(370));

    private final String zoneName;
    private final TemperatureType temperatureType;
    private final Capacity capacity;

    ZoneCode(String zoneName, TemperatureType temperatureType, Capacity capacity) {
        this.zoneName = zoneName;
        this.temperatureType = temperatureType;
        this.capacity = capacity;
    }
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/domain/enumeration/ZoneCode.kt
package com.dozycoffee.wms.warehouse.domain.enumeration

import com.dozycoffee.wms.warehouse.domain.valueobject.Capacity

enum class ZoneCode(
    val zoneName: String,
    val temperatureType: TemperatureType,
    val capacity: Capacity
) {
    A("원두", TemperatureType.AMBIENT, Capacity(180)),
    B("시럽", TemperatureType.AMBIENT, Capacity(120)),
    C("분말/파우더", TemperatureType.AMBIENT, Capacity(100)),
    D("유제품", TemperatureType.COLD, Capacity(80)),
    E("컵/소모품/포장재", TemperatureType.AMBIENT, Capacity(370))
}
